lib/src/main/java/com/dynatrace/metric/util/Metric.java

```diff
@@ -205,11 +212,36 @@ public Builder setDimensions(DimensionList dimensions) {
      * (Optional) Set the timestamp for the exported metric line. In most cases, {@link
      * Builder#setCurrentTime()} should be suitable.
      *
+     * <p>If the timestamp is from before the year 2000 or from after the year 3000 (e.g., when the
+     * wrong unit was used when creating the {@link Instant}), the timestamp will be discarded and
+     * no value will be set.
+     *
      * @param timestamp an {@link Instant} object describing the time at which the {@link Metric}
      *     was created.
      * @return this
      */
     public Builder setTimestamp(Instant timestamp) {
+      int year = timestamp.atZone(ZoneOffset.UTC).getYear();
+      if (year < 2000 || year > 3000) {
+        if (timestampWarningCounter == 0) {
+          logger.warning(
+              String.format(
+                  "Order of magnitude of the timestamp seems off. "
+                      + "The timestamp represents a time before the year 2000 or after the year 3000. "
+                      + "Skipping setting timestamp, the current server time will be added upon ingestion. "
+                      + "Only every %s of these messages will be printed.",
```

Comment:
This will result in "Only every 1000 of these messages will be printed" which sort of implies at least to me that 1000 messages are printed, but are buffered.
```suggestion
                      + "Only one out of every %s of these messages will be printed.",
```